src/warpedview.jl

```diff
@@ -0,0 +1,59 @@
+immutable WarpedView{T,N,A<:AbstractArray,F1<:Transformation,F2<:Transformation,I} <: AbstractArray{T,N}
+    parent::A
+    transform::F1
+    transform_inv::F2
+    indices::I
+
+    function (::Type{WarpedView{T,N,A,F1,F2,I}}){T,N,A<:AbstractArray,F1<:Transformation,F2<:Transformation,I}(
+            parent::A, tform::F1, tinv::F2, indices::I)
+        @assert eltype(parent) == T
+        new{T,N,A,F1,F2,I}(parent, tform, tinv, indices)
+    end
+end
+
+function WarpedView{T,N,F<:Transformation}(inner::WarpedView{T,N}, tform_in::F)
+    tform = compose(tform_in, inner.transform)
+    tinv = inv(tform)
+    etp = inner.parent
+    inds = autorange(etp, tform)
+    WarpedView{T,N,typeof(etp),typeof(tform),typeof(tinv),typeof(inds)}(etp, tform, tinv, inds)
+end
+
+function WarpedView{T,N}(parent::AbstractArray{T,N}, args...)
+    itp = Interpolations.BSplineInterpolation{T,N,typeof(parent),Interpolations.BSpline{Interpolations.Linear},OnGrid,0}(parent)
```

Comment:
Though of course if the user supplies an `itp`/`etp` object, then it's fine to consider it a view since it's a view of what's being passed in as an argument.